src/absil/ilwrite.fsi

```diff
@@ -15,17 +15,22 @@ type ILStrongNameSigner =
     static member OpenKeyPairFile: string -> ILStrongNameSigner
     static member OpenKeyContainer: string -> ILStrongNameSigner
 
+type EmitStreamProvider = Lazy<Choice<EmitTo, Diagnostic>>
```

Comment:
@dsyme ok about remove the Lazy, only question is 
```fsharp
EmittedStream of unit -> Choice<Stream,Diagnostic>
``` 
or directly
```
EmittedStream of Stream
```

i prefer the second, because if the user need a file, should use `EmittedFile`, otherwise should pass directly the `Stream`. 
If the user passa a `FileStream`/`MemoryMappedFile` with `EmittedStream` he should manage the dispose/cleanup (not needed for `MemoryStream` )
Code easier to follow, no need for lazy or deferred creation.

Note that fsc after refactoring use internally  `EmittedFile` not `EmittedStream`